react-and-redux-style-guide/README.md

```diff
@@ -0,0 +1,469 @@
+}
+```
+```typescript
+// GOOD
+import {Dispatch} from 'redux';
+export const deleteUser = (id: Identifier) => (dispatch: Dispatch, getState: RootState) => {
+    ...
+    dispatch<DeleteUserAction>(createAction(DELETE_USER, id))
+}
+```
+
+## # Do not overuse Redux, but prefer it to other approaches when possible.
+React allows us to store everything inside components, no matter if we deal with functional or class components. But this approach may lead to readability and maintainability problems when an application becomes bigger. Redux is a good place to keep an application state and updating logic in isolation from UI components. 
+
+Here are some hints when we better prefer Redux and when not.
+#### ## Use prop passing when we have up to 3 levels of children components.
+```typescript
+// EXAMPLE
+const Parent = () => {
+  const [value, setValue] = useState(1);
+  return (
+    <>
+      <Child value={value} onUpdateValue={setValue} />
+    </>
+  );
+};
+
+const Child = ({ value, onUpdateValue }) => (
+  <>
+    <GrandChild value={value} onUpdateValue={onUpdateValue} />
+  </>
+);
+
+const GrandChild = ({ value, onUpdateValue }) => (
+  ...
+);
+```  
+#### ## Prefer context when: 
+- we should pass values deeper than 3 levels down
+- there is no need to have centralized storage of state and logic
+- the size of state seems to keep small
+- state updates are infrequent 
+```typescript
+// EXAMPLE
+export const UnsavedChangesContext = createContext(null);
+
+const GroupDataContainer = (props) => {
+    let [hasUnsavedChanges, setHasUnsavedChanges] = useState(false);
+    const createUnsavedChangesContext = () => ({
+        getHasUnsavedChanges: () => hasUnsavedChanges,
+        setHasUnsavedChanges: (isChanged) => setHasUnsavedChanges(isChanged),
+    });
+
+    return (
+        <UnsavedChangesContext.Provider value={createUnsavedChangesContext()}>
+            <GroupData />
+        </HasUnsavedChangesContext.Provider>
+    );
+}
+const GroupData = () => (
+    <>
+        <HeaderLayout />
+        <GroupDataLayout />
+        <FooterLayout />
+    </>
+)
+const GroupDataLayout = () => (
+    <Block>
+        <GroupDataGrid />
+        <GridButtons />
+    </Block>
+)
+const GroupDataGrid = () => (
+    <GridLayout />
+)
+const GridLayout = () => (
+    <Block>
+        <GridHeader />
+        <GridBody />
+        <GridFooter />
+    </Block>
+)
+const GridBody = () => (
+    const {setHasUnsavedChanges} = useContext(UnsavedChangesContext);
+    
+    return (
+        <Grid onChange={setHasUnsavedChanges} />
+    );
+)
+``` 
+#### ## Redux should be used in all other cases.
+
+## # `mapStateToProps` should not bind complete store to a component
+> Reason: Passing whole state to a component is a bad practice, triggering unnecessary re-renders.
+```typescript
+// BAD
+const mapStateToProps = (state) => state
+```
+```typescript
+// GOOD
+const mapStateToProps = (state) => {isActive: state.isActive}
+```
```

Comment:
Always name selectors. `mapStateToProps` does not make sense, it should be `getState` or something like that.

Reply:
yes, I'll change the example